blackout_regex.py

```diff
@@ -1,86 +1,38 @@
 # -*- coding: utf-8 -*-
-'''
+"""
 Alerta Blackout Regex
 =====================
 
 Alerta plugin to enhance the blackout system.
-'''
-import os
+"""
 import re
-import json
-import time
-import shutil
 import logging
-import tempfile
 
-from alertaclient.api import Client
+from alerta.app import db
 from alerta.plugins import PluginBase
-from alertaclient.models.blackout import Blackout
 
-log = logging.getLogger('alerta.plugins.blackout_regex')
-
-CACHE_ENABLED = os.getenv('ALERTA_BLACKOUT_CACHE_ENABLED', True)
-CACHE_FILE = os.getenv('ALERTA_BLACKOUT_CACHE_FILE', '/var/cache/alerta/blackout_regex')
-CACHE_TIME = os.getenv('ALERTA_BLACKOUT_CACHE_TIME', 60)  # seconds
-
-client = Client()
+log = logging.getLogger("alerta.plugins.blackout_regex")
 
 
 def parse_tags(tag_list):
-    return {k: v for k, v in (i.split('=', 1) for i in tag_list if '=' in i)}
+    return {k: v for k, v in (i.split("=", 1) for i in tag_list if "=" in i)}
 
 
 class BlackoutRegex(PluginBase):
-    def _blackout_obj(self, blackouts):
-        return [Blackout.parse(blackout) for blackout in blackouts]
-
-    def _fetch_and_cache(self):
+    def _fetch_blackouts(self):
         try:
-            http_get = client.http.get('/blackouts')
-        except Exception:
-            log.error('Unable to retrieve the Blackouts from the API', exc_info=True)
-            blackouts = []
-        else:
-            blackouts = http_get['blackouts']
-            log.debug('Retrieved raw blackouts from the API:')
+            count = db.get_blackouts_count()
```

Comment:
If # of queries to the database are a concern, might want to drop the call to `get_blackouts_count` since the return value is only used for debugging. Maybe log the length of the return from `get_blackouts` instead?

Reply:
Yeah, that's not ideal...

The count is actually used in:

```python
blackouts = db.get_blackouts(page=1, page_size=count)
```

As I wasn't sure whether to gather the exact number of Blackouts, or hardcode a number (`page_size` must be a non-zero value, it seems, can't leave it blank unfortunately). Or can just use to the `QUERY_LIMIT` option (but I'm not a big fan of that option, as then we may have blackouts that aren't being evaluated due to a hidden variable).

Reply:
Ah, I missed that. Yeah, nevermind then, seems like a necessary evil :(